### PR TITLE
[WIP] Fixing #398: [new wireframe] improve home page

### LINKED ISSUE #398: [new wireframe] improve home page
- add block list
- update stats area

### PR DESCRIPTION
**Pull request by nmhung1210**
**What type of PR is this?**
ui-change

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #398 

**Special notes for your reviewer**:
This PR is still Work In Process (WIP). Which mean not ready for production merging. 

**Does this PR introduce a user-facing change?**:
This PR is changing the UI for home page according to the new wireframe. 
The new wireframe includes:

- [x] 1. Search box
- [x] 2. Stats area
- [x] 3. Block List (With Animation)

Screen Shot 2019-05-02 at 9 54 58 PM

TODO:

- [ ] Update those stats boxes with real values from the chain side.
- [ ] Feedback & finalize UI design for the home page.

### CODE REVIEW

File: src/shared/common/block-list.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+export const BlockList = () => {
+  const [latestHeight, setLatestHeight] = useState(0);
+  const [blocks, setBlocks] = useState(blockHolders);
+  const [isLoading, setLoading] = useState(true);
+
+  return (
+    <div style={{ padding: 16, backgroundColor: "#fff" }}>
+      <h3 style={{ marginBottom: 16 }}>{t("home.blockList")}</h3>
+
+      <Query query={GET_LATEST_HEIGHT}>
```
Comment (uu-z) [inline src/shared/common/block-list.tsx]: add a pollInterval (for example 10000) to GET_LATEST_HEIGHT  to poll fetch the latest blocks

File: src/shared/common/search-box.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+              const hash = get<string>(data, "getBlockMetas.blkMetas.0.hash");
+              if (hash) {
+                return history.push(`/block/${hash}`);
+              }
+            }
+          } catch (e) {
+            return history.push(`/notfound`);
+          }
+        }
+        try {
```
Comment (CoderZhi) [inline src/shared/common/search-box.tsx]: I'd prefer to do the following two asserts in the server side, or we can even wrap the whole search as one query, and then return the destination for the client side to process.

File: src/shared/home/home.tsx [hunk 3]
```diff
@@ -1,216 +1,140 @@
...
+    >
+      <h3>{"Map"}</h3>
       {
-        title: t("home.marketCap"),
-        value: `${marketCapUsd || 0} M`,
-        icon: "bank",
-        action: () => {
-          window.location.href = `https://coinmarketcap.com/currencies/iotex/?utm_source=iotexscan.i`;
-        }
+        //TODO
```
Comment (CoderZhi) [inline src/shared/home/home.tsx]: more comments of what todo

File: src/shared/home/home.tsx [hunk 4]
```diff
@@ -1,216 +1,140 @@
...
+        style={{ margin: "20px 0px", background: "#f0f2f5" }}
+      >
+        <Row gutter={GUTTER}>
+          <Col md={16}>
+            <Row gutter={GUTTER}>
+              <Col md={12}>
+                <Tile title={"TPS"} value={`0`} />
+              </Col>
+              <Col md={12}>
+                <Tile title={"# of Delegates"} value={`99`} />
```
Comment (CoderZhi) [inline src/shared/home/home.tsx]: don't forget to put these titles in translation

File: src/shared/common/block-list.tsx (hunk 1 above)
Comment (nmhung1210) [reply src/shared/common/block-list.tsx]: done

File: src/shared/home/home.tsx (hunk 3 above)
Comment (nmhung1210) [reply src/shared/home/home.tsx]: more comments added

File: src/shared/home/home.tsx (hunk 4 above)
Comment (nmhung1210) [reply src/shared/home/home.tsx]: done

File: src/shared/common/search-box.tsx (hunk 2 above)
Comment (nmhung1210) [reply src/shared/common/search-box.tsx]: I had to check

**Comment by nmhung1210** [comment]
Closed in favor of #487

**Pull request closed without merging**